packages/engine/src/worker/runner/comms/outbound.rs

```diff
@@ -57,6 +58,60 @@ impl From<flatbuffers_gen::runner_warning_generated::RunnerWarning<'_>> for Runn
         }
     }
 }
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct UserError(pub String);
+
+impl From<flatbuffers_gen::user_error_generated::UserError<'_>> for UserError {
+    fn from(user_error: flatbuffers_gen::user_error_generated::UserError<'_>) -> Self {
+        Self(user_error.msg().to_string())
+    }
+}
+
+impl std::fmt::Display for UserError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
```

Comment:
Let's forward to the display implementation.
```suggestion
        std::fmt::Display::fmt(&self.0, f)
```

Reply:
I don't really feel strongly about this, so I'm happy to change, but the `write!` method is even the [one used in the rust-by-example docs](https://doc.rust-lang.org/rust-by-example/hello/print/print_display.html)

Reply:
Also having a look through our code we much more frequently use `write!` in `Display` implementations so I'm actually going to leave it for now. If you feel strongly, or find an argument to support one over the other, lets open a ticket to change the code-base to be consistent, and then somehow capture it in some styling guidance